### PR TITLE
Set secure processing property to XML factory 

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
- set secure processing property to xml factory and parser
- port xml utils changes from v1.

### CODE REVIEW

File: core/src/main/java/software/amazon/awssdk/core/util/XpathUtils.java [hunk 1]
```diff
@@ -624,4 +625,17 @@ private static String evaluateAsString(String expression, Node node,
     private static boolean isEmptyString(String s) {
         return s == null || s.trim().length() == 0;
     }
+
+    /**
+     * Disables certain dangerous features that attempt to automatically fetch DTDs
+     *
+     * See <a href="https://www.owasp.org/index.php/XML_External_Entity_(XXE)_Prevention_Cheat_Sheet">OWASP XXE Cheat Sheet</a>
+     */
+    private static DocumentBuilderFactory documentBuilderFactory() throws ParserConfigurationException {
```
Comment (spfink) [inline core/src/main/java/software/amazon/awssdk/core/util/XpathUtils.java]: Can this also go in XmlUtils?

File: test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/asserts/marshalling/XmlAsserts.java [hunk 2]
```diff
@@ -71,11 +73,18 @@ private static String formatXml(String xmlDocumentString) throws Exception {
...
-        Transformer transformer = TransformerFactory.newInstance().newTransformer();
+        Transformer transformer = transformerFactory().newTransformer();
         transformer.setOutputProperty(OutputKeys.INDENT, "yes");
         StreamResult result = new StreamResult(new StringWriter());
         DOMSource source = new DOMSource(xmlDocument);
         transformer.transform(source, result);
         return result.getWriter().toString();
     }
+
+    private static TransformerFactory transformerFactory() throws TransformerConfigurationException {
```
Comment (spfink) [inline test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/asserts/marshalling/XmlAsserts.java]: XmlUtils?

Comment (zoewangg) [reply test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/asserts/marshalling/XmlAsserts.java]: This is only used in unit tests to format the xml and not expected to be used anywhere else. I'll move the other one to `XmlUtils`

**Comment by zoewangg** [comment]
Ran integration-tests against S3 and other xml-protocol services and all tests passed.

**Pull request merged**